src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateGenerator.cs

```diff
@@ -244,11 +250,21 @@ private X509CertificateContainer CreateCertificate(bool isAuthority, bool isMach
 
             if (!isAuthority && string.IsNullOrWhiteSpace(subjects[0]))
             {
-                throw new ArgumentException("Certificate Subject must not be an empty string or only whitespace", "subjects");
+                throw new ArgumentException("Certificate Subject must not be an empty string or only whitespace", "creationSettings.Subjects");
             }
 
             EnsureInitialized();
 
+            // Set to default cert creation settings if not set
+            if (certificateCreationSettings.ValidityNotBefore == default(DateTime))
+            {
+                certificateCreationSettings.ValidityNotBefore = _validityNotBefore;
+            }
+            if (certificateCreationSettings.ValidityNotAfter == default(DateTime))
+            {
+                certificateCreationSettings.ValidityNotAfter = _validityNotAfter;
```

Comment:
I feel like these checks belong just after `if (certificateCreationSettings == null)`. Basically if we detect that `certificateCreationSettings == null`, we should create the new certificateCreationSettings object and initialize the values there.